framework/Models/PostProcessor.py

```diff
@@ -99,34 +105,54 @@ def specializeValidateDict(cls):
     cls.validateDict['Function'  ][0]['required'    ] = False
     cls.validateDict['Function'  ][0]['multiplicity'] = 1
 
-  def __init__(self):
+  def __init__(self ):
     """
       Constructor
-      @ In, runInfoDict, dict, the dictionary containing the runInfo (read in the XML input file)
+      @ In, None
       @ Out, None
     """
     super().__init__()
     self.inputCheckInfo  = []     # List of tuple, i.e input objects info [('name','type')]
     self.action = None            # action
-    self.workingDir = ''          # path for working directory
     self.printTag = 'POSTPROCESSOR MODEL'
-    self.outputDataset  = False # True if the user wants to dump the outputs to dataset
-    self.validDataType = ['PointSet','HistorySet'] # The list of accepted types of DataObject
-    ## Currently, we have used both DataObject.addRealization and DataObject.load to
-    ## collect the PostProcessor returned outputs. DataObject.addRealization is used to
-    ## collect single realization, while DataObject.load is used to collect multiple realizations
-    ## However, the DataObject.load can not be directly used to collect single realization
-    ## One possible solution is all postpocessors return a list of realizations, and we only
-    ## use addRealization method to add the collections into the DataObjects
-    self.outputMultipleRealizations = False
-
-  def _handleInput(self, paramInput):
-    """
-      Function to handle the common parts of the model parameter input.
-      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+    self._pp = None
+
+  def _readMoreXML(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
       @ Out, None
     """
-    Model._handleInput(self, paramInput)
+    Model._readMoreXML(self, xmlNode)
```

Comment:
why the readMoreXML?

Reply:
this should disappear

Reply:
there was a lot of talk about this in the Entity/Interface PR; unfortunately, the conversion to purely using `handleInput` is more than just adding `handleInput`; the `Simulation` is set up to only purely use `handleInput` if a particular flag is thrown by the module (or factory), which requires the module (or factory) to create the input specs for the object. This only exists for a very few objects in RAVEN (the Database I think is the only one I recall offhand).

In short, some significant work needs to be done to eliminate `readMoreXML` given how `Simulation` interacts with the modules.

Reply:
This is needed to handle Assembler correctly. I think we really need to clean up our code to remove readMoreXML, otherwise, you will see this legacy method called in several places.

Reply:
Okay for now, this is a priority.